src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -542,7 +549,7 @@ where
     // with other global Cargos
     let _lock = config.acquire_package_cache_lock()?;
 
-    if needs_update {
+    if let NeedsUpdate::True = needs_update {
         source.update()?;
```

Comment:
Since an update is now happening twice, could this update move to closer where the other update is happening? It feels a bit odd to have these two updates so far apart.